### DISCUSSION TITLE
Webhook config

### DISCUSSION THREAD

**Discussion by ivanmartinezjurjo** [netbox-community/netbox#18166] [Q&A] [answered]
{
  "chat_id": "****", 
  "message_thread_id": "****",
  "text": "🚨 *Updated virtual machine*\n
  \n*Name:* 🏷️ {{ data.name }}
  \n*Status:* {% if data.status.label == 'Active' %}🟢 {{ data.status.label }}{% else %}🔴 {{ data.status.label }}{% endif %}
  \n*Cluster:* 🖧 {{ data.cluster.name }}
  \n*IP:* 🌐 {{ data.primary_ip4.address }}
  \n*OS:* 💻 {{ data.platform.name | default('No especificado') }}
  \n🔄 *Modified fields:* ✏️ {{ changed_fields | join(', ') | default('No hay cambios detectados') }}",
  "parse_mode": "Markdown"
}

I made this for a telegram alert and everything its ok except the field  "Modified fields" that always pick the default response, y tried using previous.data to compare other fields but never it never takes the changes.
If somebody knows if maytbe that isnt the netbox standard variable or where i can get a list of them, i will truly appreciate it, i'm a bit stucked here.

**Comment by candlerb** [comment, marked as answer]
> If somebody knows if maytbe that isnt the netbox standard variable or where i can get a list of them

Here is the list: https://netboxlabs.com/docs/netbox/en/stable/integrations/webhooks/#available-context

There's no value called "changed_fields". But there are `snapshots["prechange"]` and `snapshots["postchange"]`. You could iterate over these and compare them. Note that `snapshots["prechange"]` can be null for a newly-created object, and I think `snapshots["postchange"]` can be null for a deleted object.

**Reply by candlerb** [reply]
Please edit your post, and put three backticks (` ``` `) on a line of their own, before and after the jinja2 code block. This will stop Markdown interpreting the content as emojis.
